common/primitives/core/src/omni/intent.rs

```diff
@@ -4,23 +4,30 @@ use scale_info::TypeInfo;
 use sp_core::H160;
 use sp_runtime::{traits::ConstU32, BoundedVec};
 
-pub const CALL_ETHEREUM_INPUT_LEN: u32 = 10 * 1024;
-pub const MAX_REMARK_LEN: u32 = u32::max_value();
+pub type CallEthereumInputLen = ConstU32<{ 10 * 1024 }>;
+pub type RemarkLen = ConstU32<{ u32::max_value() }>;
+pub type IntentStringLen = ConstU32<128>;
 
-#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, TypeInfo)]
+pub type IntentId = u32;
+
+#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
 pub enum Intent {
     #[codec(index = 0)]
-    CrossChainSwap(SwapOrder),
-    #[codec(index = 1)]
     TransferEthereum(TransferEthereum),
-    #[codec(index = 2)]
+    #[codec(index = 1)]
     CallEthereum(CallEthereum),
+    #[codec(index = 2)]
+    SystemRemark(BoundedVec<u8, RemarkLen>),
     #[codec(index = 3)]
-    SystemRemark(BoundedVec<u8, ConstU32<MAX_REMARK_LEN>>),
-    #[codec(index = 4)]
     TransferNative(TransferNative),
-    #[codec(index = 5)]
+    #[codec(index = 4)]
     TransferSolana(TransferSolana),
+    #[codec(index = 5)]
+    Swap(
+        SwapOrder,
+        Option<CrossChainSwapProvider>,
+        SingleChainSwapProvider,
```

Comment:
Should we merge these two? We could use an enum like:

```rust
enum SwapProvider {
  CrossChain(CrossChainSwapProvider)
  SingleChain(SingleChainSwapProvider) 
}
```
Will the FE (or user) choose the provider?

Reply:
That's my imagination (in the future)

Reply:
Yes - it's not an `Option` right now, we always need a `SingleChainSwapProvider` to process swaps on the dest-chain

Shall I rename it to `DestChainSwapProvider`?

Reply:
oh I see, I understand now. It sounds good